### ISSUE TITLE
Some sort of circular reference

### ISSUE THREAD

**Issue by TheDenSter** [microsoft/AL#6119]
**Describe the bug**
I created a new codeunit 50000 GenJnlPostLineSubscribers and as soon as I give it its proper name and save it, I get two problems:
- An application object of type 'Codeunit' with ID '23029100' is already declared by the extension 'MyExtension by MyPublisher (1.0.0.0)'AL(AL0264)
- An application object of type 'Codeunit' with name 'GenJnlPostLineSubscribers' is already declared by the extension 'MyExtension by MyPublisher (1.0.0.0)'AL(AL0197)

There is only one codeunit with that number and name. As a result, intellisense references in EventSubscriber procedure attributes no longer work and I get a request to create a new issue in the output:
```
[Error - 5:24:56 PM] Please report this issue to https://github.com/microsoft/al/issues including information on how to reproduce it, if possible.
Processing of message 'textDocument/completion' failed with error: 'Node span: 0-65, position: 105 (Parameter 'position')'
Details:
System.ArgumentOutOfRangeException: Node span: 0-65, position: 105 (Parameter 'position')
   at Microsoft.Dynamics.Nav.CodeAnalysis.SyntaxNode.FindToken(Int32 position, Boolean findInsideTrivia) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Syntax\SyntaxNode.cs:line 1002
   at Microsoft.Dynamics.Nav.CodeAnalysis.SyntaxNode.FindToken(Int32 position, Func`2 findInsideTrivia) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Syntax\SyntaxNode.cs:line 1035
   at Microsoft.Dynamics.Nav.CodeAnalysis.SyntaxNode.FindToken(Int32 position, Boolean findInsideTrivia) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Syntax\SyntaxNode.cs:line 1002
   at Microsoft.Dynamics.Nav.CodeAnalysis.Syntax.SyntaxTreeExtensions.IsEntirelyWithinStringLiteral(SyntaxTree syntaxTree, Int32 position, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Syntax\SyntaxTreeExtensions.cs:line 515
   at Microsoft.Dynamics.Nav.CodeAnalysis.Syntax.SyntaxTreeExtensions.IsInNonUserCode(SyntaxTree syntaxTree, Int32 position, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Syntax\SyntaxTreeExtensions.cs:line 221
   at Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces.Completion.MemberSyntaxContext.CreateContext(Workspace workspace, SemanticModel semanticModel, Int32 position, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces\Completion\ContextQuery\MemberSyntaxContext.cs:line 173
   at Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces.Completion.CompletionProvider.CreateMemberContextAsync(Document document, Int32 position, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces\Completion\CompletionProvider.cs:line 82
   at Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces.Completion.AbstractSymbolCompletionProvider.GetPerContextSymbols(Document document, Int32 position, OptionSet options, IEnumerable`1 relatedDocuments, Boolean preselect, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces\Completion\AbstractSymbolCompletionProvider.cs:line 248
   at Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces.Completion.AbstractSymbolCompletionProvider.GetItemsWorkerAsync(Document document, AbstractSyntaxContext context, TextSpan span, OptionSet options, Boolean preselect, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces\Completion\AbstractSymbolCompletionProvider.cs:line 199
   at Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces.Completion.AbstractSymbolCompletionProvider.ProvideCompletionsAsync(CompletionContext context, AbstractSyntaxContext memberSyntaxContext) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces\Completion\AbstractSymbolCompletionProvider.cs:line 174
   at Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces.Completion.CompletionServiceWithProviders.GetContextAsync(CompletionProvider provider, Document document, Int32 position, CompletionTrigger triggerInfo, OptionSet options, Nullable`1 defaultSpan, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces\Completion\CompletionServiceWithProviders.cs:line 372
   at Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces.Completion.CompletionServiceWithProviders.GetCompletionsAsync(Document document, Int32 caretPosition, CompletionTrigger trigger, ImmutableHashSet`1 roles, OptionSet options, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis.Workspaces\Completion\CompletionServiceWithProviders.cs:line 183
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.LanguageServer.CompletionRequestHandler.HandleAsync(TextDocumentPositionParams request, Int32 requestId, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\LanguageServer\CompletionRequestHandler.cs:line 45
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.MessageProtocol.RequestHandlerBase`1.HandleAsync(JToken requestContents, Int32 requestId, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\MessageProtocol\RequestHandlerBase.cs:line 54
   at Microsoft.Dynamics.Nav.EditorServices.Protocol.RequestRegistry.Process(Message message) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.EditorServices.Protocol\Endpoints\RequestRegistry.cs:line 80
```

**Expected behavior**
I expect not to get an error message that the codeunit that I just created already exists, when there is only one copy

**5. Versions:**
![image](https://user-images.githubusercontent.com/23434319/91919143-d9f2fe00-ec79-11ea-8362-586d5ae2704a.png)
This is running in a brand new sandbox container, created this morning

**Comment by thpeder** [comment]
HI @TheDenSter,
Thanks for reporting the issue, but I'm unable to reproduce it.
If you are able to reproduce it again, could you record a video/gif with "Screencast Mode" toggled on in VSCode, and explain how you got to that issue.

**Comment by bqs-lks** [comment]
I believe this is the same issue as #6077.

![image](https://user-images.githubusercontent.com/59433306/91965895-b4cac200-ed43-11ea-8243-88a353fcc04e.png)

Hope this helps :)

**Comment by TheDenSter** [comment]
Yes that does look like a similar issue, I'm glad I'm not going crazy :). I completely rebuilt the entire workspace and I did not save the old one. Now I can't even repro this myself.

Searching for the codeunit by its name it says there are two files. One of them was in the 'src\codeunit' folder and another was in the 'src\Codeunit' folder (note that one starts with lower case c and the other with upper case C). VSCode THINKS it is two files but it is actually just one. When I showed the files side by side and made a change to the one, that change was also made in the other. Makes sense because it was really just one file.

I'll just close this, because I don't have the workspace anymore and it's not happening in the new one.
